### PR TITLE
Refactor HgiGateway status to use async state model (#537)

### PR DESCRIPTION
**Pull request by PWhite-Eng**
### The Problem:

Currently, determining if the Gateway is active relies on synchronous engine querying which bypasses the standard asynchronous state model of the architecture. This technical debt makes state tracking inconsistent and harder to maintain for Home Assistant integrations (Issue #537).

### Consequences:

If left unaddressed, the library retains architectural fragmentation by mixing synchronous attribute polling with async state evaluations. This limits the ability to safely track and expose the Gateway's live connection status to downstream consumers (like ramses_cc), potentially leading to stale UI representations or blocking calls.

### The Fix:

We encapsulated the gateway state logic within the HgiGateway class itself by adding an asynchronous is_active() method. We also introduced a formal configurable timeout to define exactly when a gateway is considered "inactive".

### Technical Implementation:
- **const.py**: Added GATEWAY_MESSAGE_TIMEOUT = timedelta(minutes=5) to define a standard heartbeat threshold.
- **base.py**: Added async def is_active(self) -> bool: to the HgiGateway class. It fetches self._gwy._engine._this_msg, strictly checks the datetime (dtm), and calculates if the time delta against now(UTC) is within the timeout window. Handles both naive and timezone-aware datetimes.
- **test_base.py**: Combined scattered base tests into a unified test file. Wrote explicit tests verifying behavior for missing messages, active messages, expired messages, naive datetimes, and mock component behaviors (DeviceBase, BatteryState). Mypy strict compliance was enforced using getattr() fallbacks and __class__ comparisons to prevent static type narrowing errors.

Note: This PR introduces a breaking change to how gateway status is queried and **MUST** be merged at the exact same time as the matching PR for ramses_cc.

### Testing Performed:
- Executed the full suite of pytest tests (805 passed, 0 failures).
- Achieved 100% test coverage on the newly implemented gateway status logic.
- Executed mypy --strict with zero issues found across 131 source files.
- Addressed boundary cases around simulated "Faked" devices, battery mixins, and strict type checking edge-cases.

### Risks of NOT Implementing:

Failing to adopt this change leaves the gateway state tracking coupled to internal engine attributes, making future architectural shifts difficult and maintaining legacy synchronous blocks in an asynchronous application.

### Risks of Implementing:

Consumers of the ramses_rf library (specifically ramses_cc) that currently inspect the gateway engine properties directly or rely on synchronous evaluation will break.

### Mitigation Steps:

We have ensured complete backward compatibility for standard devices while explicitly isolating the breaking changes to HgiGateway.is_active(). A corresponding branch and PR for ramses_cc has been developed in tandem to utilize resolve_async_attr to handle this new implementation gracefully.

### AI Assistance Disclosure:

This contribution was developed with the assistance of Google Gemini 3.1 Pro for code generation and documentation. No Agentic AI systems were employed; all logic and implementations were reviewed, verified, and manually committed by the author.

### CODE REVIEW

**Comment by PWhite-Eng** [comment]
@silverailscolo, this PR is stacked ontop of #539, please merge that PR before merging this PR. Please note there is a matching PR for ramses_cc to be released shortly

File: src/ramses_rf/device/heat.py [hunk 1]
```diff
@@ -5,13 +5,16 @@
 
 import logging
 from collections.abc import Callable
+from datetime import timedelta
```
Comment (silverailscolo) [inline src/ramses_rf/device/heat.py:8]: Also here: `import timedelta as td`

File: src/ramses_rf/device/hvac.py [hunk 2]
```diff
@@ -5,10 +5,13 @@
 
 import logging
 from collections.abc import Callable
+from datetime import timedelta
```
Comment (silverailscolo) [inline src/ramses_rf/device/hvac.py:8]: `as td`

File: src/ramses_rf/device/base.py [hunk 3]
```diff
@@ -88,6 +96,32 @@ def __lt__(self, other: object) -> bool:
             return NotImplemented
         return self.id < other.id  # type: ignore[no-any-return]
 
+    @property
+    def heartbeat_timeout(self) -> td:
+        """Return the timeout before the device is considered unavailable.
+
+        :return: The timeout duration before going unavailable.
+        :rtype: timedelta
```
Comment (silverailscolo) [inline src/ramses_rf/device/base.py:104]: td?

File: src/ramses_rf/device/heat.py [hunk 4]
```diff
@@ -720,6 +729,15 @@ def __init__(
         self._msgs_ot: dict[MsgId, Message] = {}
         # lf._msgs_ot_ctl_polled = {}
 
+    @property
+    def heartbeat_timeout(self) -> timedelta:
```
Comment (silverailscolo) [inline src/ramses_rf/device/heat.py:733]: `-> td`

File: src/ramses_rf/device/heat.py [hunk 5]
```diff
@@ -720,6 +729,15 @@ def __init__(
...
+        """Return the timeout before the device is considered unavailable.
+
+        :return: The timeout duration.
+        :rtype: timedelta
```
Comment (silverailscolo) [inline src/ramses_rf/device/heat.py:737]: 'td'

File: src/ramses_rf/device/heat.py [hunk 6]
```diff
@@ -1491,6 +1509,15 @@ class TrvActuator(BatteryState, HeatDemand, Setpoint, Temperature):  # TRV (04):
     _SLUG = DevType.TRV
     _STATE_ATTR = SZ_HEAT_DEMAND
 
+    @property
+    def heartbeat_timeout(self) -> timedelta:
```
Comment (silverailscolo) [inline src/ramses_rf/device/heat.py:1513]: `-> td`

File: src/ramses_rf/device/heat.py [hunk 7]
```diff
@@ -1491,6 +1509,15 @@ class TrvActuator(BatteryState, HeatDemand, Setpoint, Temperature):  # TRV (04):
...
+        """Return the timeout before the device is considered unavailable.
+
+        :return: The timeout duration.
+        :rtype: timedelta
```
Comment (silverailscolo) [inline src/ramses_rf/device/heat.py:1517]: `td`

File: src/ramses_rf/device/hvac.py [hunk 8]
```diff
@@ -95,7 +105,14 @@ class HvacSensorBase(DeviceHvac):
     It provides common functionality for sensor data collection and processing.
     """
 
-    pass
+    @property
+    def heartbeat_timeout(self) -> timedelta:
+        """Return the timeout before the device is considered unavailable.
+
+        :return: The timeout duration.
+        :rtype: timedelta
```
Comment (silverailscolo) [inline src/ramses_rf/device/hvac.py:113]: `td`

File: tests/tests_rf/device/test_base.py [hunk 9]
```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+"""Unittests for the base device classes and HgiGateway.
+
+This module combines tests for DeviceBase, HgiGateway, and BatteryState
+which all reside in ramses_rf/device/base.py.
+"""
+
+from datetime import UTC, datetime, timedelta
```
Comment (silverailscolo) [inline tests/tests_rf/device/test_base.py:8]: as dt, td

**Review by silverailscolo — CHANGES_REQUESTED** [review]
Fine to use this. Might store the gwy latest_dtm in a variable, but this also handles all Entities.
Please update `datetime` import style including in tests

File: tests/tests_rf/device/test_base.py [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+from unittest.mock import MagicMock
+
+import pytest
+
+from ramses_rf.const import GATEWAY_MESSAGE_TIMEOUT
```
Comment (silverailscolo) [inline tests/tests_rf/device/test_base.py:13]: mypy flags this as `GATEWAY_MESSAGE_TIMEOUT` missing in ramses_rf/const.py (indeed it isn't in master yet)

File: src/ramses_rf/device/base.py [hunk 11]
```diff
@@ -15,6 +15,7 @@
 from ramses_rf.binding_fsm import BindingManager, Vendor
 from ramses_rf.const import (
     DEV_TYPE_MAP,
+    GATEWAY_MESSAGE_TIMEOUT,
```
Comment (silverailscolo) [inline src/ramses_rf/device/base.py:18]: mypy error: GATEWAY_MESSAGE_TIMEOUT is not in const.py

Comment (silverailscolo) [reply src/ramses_rf/device/base.py:18]: It is own master fine. Action is looking at branch, ignore

File: tests/tests_rf/device/test_base.py (hunk 10 above)
Comment (silverailscolo) [reply tests/tests_rf/device/test_base.py:13]: NP, see below

**Comment by silverailscolo** [comment]
GATEWAY_MESSAGE_TIMEOUT: td = td(minutes=5) missing from const, will fix

**Review by silverailscolo — APPROVED** [review]
